integrations/gitlab_v2/main.py

```diff
@@ -0,0 +1,157 @@
+import typing
+from enum import StrEnum
+from typing import Any
+
+from loguru import logger
+from port_ocean.context.event import event
+from port_ocean.context.ocean import ocean
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+
+from client import CREATE_UPDATE_WEBHOOK_EVENTS, DELETE_WEBHOOK_EVENTS, GitlabClient
+from integration import GitlabProjectResourceConfig
+from utils import extract_issue_payload, extract_merge_request_payload
+
+
+class InvalidTokenException(Exception):
+    ...
+
+
+class ResourceKind(StrEnum):
+    GROUP = "group"
+    PROJECT = "project"
+    MERGE_REQUEST = "merge_request"
+    ISSUE = "issue"
+
+
+class WebHookEventType(StrEnum):
+    MERGE_REQUEST = "merge_request"
+    ISSUE = "issue"
+
+
+# Listen to the start event of the integration. Called once when the integration starts.
+@ocean.on_start()
+async def on_start() -> None:
+    logger.info("Starting musah_gitlab integration")
+    tokens = ocean.integration_config["gitlab_access_tokens"]["tokens"]
+
+    # Check if tokens is a list and filter valid tokens (strings only)
+    if isinstance(tokens, list):
+        tokens_are_valid = filter(lambda token: isinstance(token, str), tokens)
+
+        # Ensure all tokens are valid strings
+        if not all(tokens_are_valid):
+            raise InvalidTokenException("Invalid access tokens, ensure all tokens are valid strings")
+    else:
+        raise InvalidTokenException("Invalid access tokens, confirm you passed in a list of tokens")
+
+    return await bootstrap_client()
+
+
+def initialize_client(gitlab_access_token: str) -> GitlabClient:
+    return GitlabClient(
+        ocean.integration_config["gitlab_host"],
+        gitlab_access_token,
+    )
+
+
+async def bootstrap_client() -> None:
+    app_host = ocean.integration_config["app_host"]
+    tokens = ocean.integration_config["gitlab_access_tokens"]["tokens"]
+    if not app_host:
+        logger.warning(
+            "No app host provided, skipping webhook creation. "
+            "Without setting up the webhook, the integration will not export live changes from Gitlab"
+        )
+        return
+
+    if ocean.event_listener_type == "ONCE":
+        logger.info("Skipping webhook creation because the event listener is ONCE")
+        return
+
+    gitlab_client = initialize_client(tokens[0])
```

Comment:
can you share how the token is passed to the integration?

Reply:
The tokens can be passed as an environment variable 
`OCEAN__INTEGRATION__CONFIG__GITLAB_ACCESS_TOKENS={"tokens": ["token1", "token2"]}`

Also validated in the `spec.yaml` like so:

> configurations:
  - name: gitlabAccessTokens
    required: true
    type: object
    sensitive: true
    description: 'Gitlab access token. See the <a target="_blank" href="https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html">Gitlab Documentation</a>'